test/test_inference_flows.py

```diff
@@ -0,0 +1,246 @@
+        )
+        compiled_dynamic_fp8_mlp = torch.compile(
+            dynamic_fp8_mlp, backend=compile_backend, fullgraph=True
+        )
+
+        self.base_test_mlp_transform(
+            compiled_original_mlp, compiled_dynamic_fp8_mlp, input_tensor
+        )
+
+    @pytest.mark.parametrize("compile_backend", ["eager", "inductor"])
+    @pytest.mark.parametrize("dtype", [torch.bfloat16, torch.float32])
+    @unittest.skipIf(
+        not torch.cuda.is_available() or not is_H100,
+        "CUDA not available or on non H100 machine",
+    )
+    def test_static_fp8_mlp(self, compile_backend, dtype):
+        original_mlp = FeedForward().to("cuda", dtype=dtype)
+        original_mlp.reset_parameters()
+
+        static_fp8_mlp = copy.deepcopy(original_mlp)
+        quant_config = QuantConfig(
+            ActivationCasting.STATIC,
+            static_quantization_scale=torch.tensor(
+                [1.0], device="cuda", dtype=torch.float32
+            ),
+        )
+        quantize_to_float8(static_fp8_mlp, quant_config)
+
+        batch_size = 4
+        num_tokens = 1024
+        embedding_dim = 4096
+
+        input_tensor = torch.randn(
+            batch_size, num_tokens, embedding_dim, device="cuda", dtype=dtype
+        )
+
+        # Compile the models
+        compiled_original_mlp = torch.compile(
+            original_mlp, backend=compile_backend, fullgraph=True
+        )
+        compiled_static_fp8_mlp = torch.compile(
+            static_fp8_mlp, backend=compile_backend, fullgraph=True
+        )
+
+        self.base_test_mlp_transform(
+            compiled_original_mlp, compiled_static_fp8_mlp, input_tensor
+        )
+
+    @pytest.mark.parametrize("compile_backend", ["eager", "inductor"])
+    @pytest.mark.parametrize("dtype", [torch.bfloat16, torch.float32])
+    @unittest.skipIf(
+        not torch.cuda.is_available() or not is_H100,
+        "CUDA not available or on non H100 machine",
+    )
+    def test_weight_only_fp8_mlp(self, compile_backend, dtype):
+        original_mlp = FeedForward().to("cuda", dtype=dtype)
+        original_mlp.reset_parameters()
+
+        static_fp8_mlp = copy.deepcopy(original_mlp)
+        quant_config = QuantConfig(ActivationCasting.WEIGHT_ONLY)
+        quantize_to_float8(static_fp8_mlp, quant_config)
+
+        batch_size = 4
+        num_tokens = 1024
+        embedding_dim = 4096
+
+        input_tensor = torch.randn(
+            batch_size, num_tokens, embedding_dim, device="cuda", dtype=dtype
+        )
+
+        # Compile the models
+        compiled_original_mlp = torch.compile(
+            original_mlp, backend=compile_backend, fullgraph=True
+        )
+        compiled_static_fp8_mlp = torch.compile(
+            static_fp8_mlp, backend=compile_backend, fullgraph=True
+        )
+
+        self.base_test_mlp_transform(
+            compiled_original_mlp, compiled_static_fp8_mlp, input_tensor
+        )
+
+
+class TestFP8TrainToFP8LinearInference:
```

Comment:
optional: I think we could delete this if you want to lighten the support load, since we already test float8 training is close to high precision, and float8 inference is close to high precision